Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -361,4 +433,46 @@ public class CRTClientEngine: HTTPClient {
         response.body = .stream(stream)
         return requestOptions
     }
+
+    actor ContinuationFlag {
+        private var continuationResumed = false
+        public func getContinuationResumedFlag() -> Bool {
+            return continuationResumed
+        }
+        public func setContinuationResumedFlag(_ val: Bool) {
+            continuationResumed = val
+        }
+    }
+
+    private func safeResumeThrowContinuation (
+        continuationFlagSyncQueue: DispatchQueue,
+        continuationFlag: ContinuationFlag,
+        continuation: StreamContinuation,
+        error: Error
+    ) async {
+        continuationFlagSyncQueue.async {
+            Task {
```

Comment:
Getting onto a DispatchQueue just to start a task has no effect whatsoever.  The async queue task will return immediately after the Task is created & before it finishes.

Might as well start the Task from the calling thread without the dispatch queue, the effect will be identical.